apps/financial-aid/web-osk/src/routes/application/RelationshipstatusForm/RelationshipstatusForm.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState, useContext } from 'react'
+import { Text } from '@island.is/island-ui/core'
+
+import {
+  ContentContainer,
+  Footer,
+  RadioButtonContainer,
+  SpouseInfo,
+} from '@island.is/financial-aid-web/osk/src/components'
+
+import { useRouter } from 'next/router'
+
+import useFormNavigation from '@island.is/financial-aid-web/osk/src/utils/useFormNavigation'
+import cn from 'classnames'
+import * as styles from './relationshipstatusForm.treat'
+
+import {
+  FamilyStatus,
+  isEmailValid,
+  NavigationProps,
+} from '@island.is/financial-aid/shared/lib'
+import { UserContext } from '@island.is/financial-aid-web/osk/src/components/UserProvider/UserProvider'
+
+const RelationshipstatusForm = () => {
+  const router = useRouter()
+
+  const { user, setUser } = useContext(UserContext)
+
+  const navigation: NavigationProps = useFormNavigation(
+    router.pathname,
+  ) as NavigationProps
+
+  const [hasError, setHasError] = useState(false)
+
+  const [acceptData, setAcceptData] = useState(false)
+
+  const options = [
+    {
+      label: 'Nei, ég er ekki í sambúð',
+      value: FamilyStatus.UNKNOWN,
+    },
+    {
+      label: 'Já, ég er í óstaðfestri sambúð',
+      value: FamilyStatus.UNREGISTEREDCOBAHITATION,
+    },
+  ]
+
+  const errorCheck = () => {
+    if (!user?.familyStatus) {
+      setHasError(true)
+      return
+    }
+
+    if (!navigation?.nextUrl) {
+      return
+    }
+
+    if (user?.familyStatus === FamilyStatus.UNKNOWN) {
+      router.push(navigation?.nextUrl)
+    }
+
+    if (user?.familyStatus === FamilyStatus.UNREGISTEREDCOBAHITATION) {
+      if (!acceptData || !user.spouse?.email || !user.spouse?.nationalId) {
+        setHasError(true)
+        return
+      }
+
+      if (
+        !isEmailValid(user.spouse?.email) ||
+        user.spouse?.nationalId.length !== 10
+      ) {
+        setHasError(true)
+        return
+      }
+
+      router.push(navigation?.nextUrl)
+    }
+  }
+
+  return (
+    <>
+      <ContentContainer>
+        <Text as="h1" variant="h2" marginBottom={2}>
+          Hjúskaparstaða þín
+        </Text>
+
+        <Text marginBottom={[3, 3, 4]}>
+          Samkvæmt upplýsingum frá Þjóðskrá ert þú ekki í staðfestri sambúð. En
+          sért þú í óstaðfestri sambúð þarft þú og maki þinn bæði að skila inn
+          umsókn um fjárhagsaðstoð.
+        </Text>
+
+        <Text as="h3" variant="h3" marginBottom={[3, 3, 4]}>
+          Ert þú í óstaðfestri sambúð?
+        </Text>
+
+        <RadioButtonContainer
+          options={options}
+          error={hasError && !user?.familyStatus}
+          isChecked={(value: FamilyStatus) => {
+            return value === user?.familyStatus
+          }}
+          onChange={(value: FamilyStatus) => {
+            if (user) {
```

Comment:
What happens if user is undefined in the component? Will that ever happen? Just thinking if we should check for it above instead of checking every time.